.circleci/config.yml

```diff
@@ -1,23 +1,402 @@
+    parameters:
+      filename:
+        type: string
+    steps:
+      - store_artifacts:
+          path: ~/mattermost-mobile/<<parameters.filename>>
 
 jobs:
   test:
     working_directory: ~/mattermost-mobile
     docker:
       - image: circleci/node:10
     steps:
-      - checkout
-      - run: |
-          echo assets/base/config.json
-          cat assets/base/config.json
-          # Avoid installing pods
-          touch .podinstall
-          # Run tests
-          make test || exit 1
+      - checkout:
+          path: ~/mattermost-mobile
+      - npm-dependencies
+      - assets
+      - run:
+          name: Check styles
+          command: npm run check
+      - run:
+          name: Running Tests
+          command: npm test
+
+  build-android-beta:
+    executor: android
+    environment: # remove env vars on final PR
+      SENTRY_ENABLED: false
+      AWS_FOLDER_NAME: ''
+      MATTERMOST_WEBHOOK_URL: ''
+    steps:
+      - build-android
+      - persist
+      - save:
+          filename: "Mattermost_Beta.apk"
+
+  build-android-release:
+    executor: android
+    environment: # remove env vars on final PR
+      SENTRY_ENABLED: false
+      AWS_FOLDER_NAME: ''
+      MATTERMOST_WEBHOOK_URL: ''
+    steps:
+      - build-android
+      - persist
+      - save:
+          filename: "Mattermost.apk"
+
+  build-android-pr:
+    executor: android
+    environment:
+      BRANCH_TO_BUILD: ${CIRCLE_BRANCH}
+    steps:
+      - build-android
+      - persist
+      - save:
+          filename: "Mattermost_Beta.apk"
+
+  build-ios-beta:
+    executor: ios
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost_Beta.ipa"
+
+  build-ios-release:
+    executor: ios
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost.ipa"
+
+  build-ios-pr:
+    executor: ios
+    environment:
+      BRANCH_TO_BUILD: ${CIRCLE_BRANCH}
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost_Beta.ipa"
+
+  deploy-android-release:
+    executor:
+      name: android
+      resource_class: medium
+    steps:
+      - deploy-android:
+          apk_path: Mattermost.apk
+
+  deploy-android-beta:
+    executor:
+      name: android
+      resource_class: medium
+    steps:
+      - deploy-android:
+          apk_path: Mattermost_Beta.apk
+
+  deploy-ios-release:
+    executor: ios
+    steps:
+      - deploy-ios:
+          ipa_path: Mattermost.ipa
+
+  deploy-ios-beta:
+    executor: ios
+    steps:
+      - deploy-ios:
+          ipa_path: Mattermost_Beta.ipa
 
 workflows:
   version: 2
-  pr-test:
+  build:
     jobs:
       - test
+
+      - build-android-release:
+          context: mattermost-mobile-android-release
+          requires:
+            - test
+          filters:
+            branches:
+              only:
+                - /^build-[0-9]*$/
+                - /^build-(android).*$/
```

Comment:
same as on ios, why using parens, if you are not using any other operator is not adding any value